e2e/events/events.go

```diff
@@ -0,0 +1,182 @@
+package events
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/nomad/e2e/e2eutil"
+	"github.com/hashicorp/nomad/e2e/framework"
+	"github.com/hashicorp/nomad/helper/uuid"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/hashicorp/nomad/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+type EventsTest struct {
+	framework.TC
+	jobIDs []string
+}
+
+func init() {
+	framework.AddSuites(&framework.TestSuite{
+		Component:   "Events",
+		CanRunLocal: true,
+		Cases: []framework.TestCase{
+			new(EventsTest),
+		},
+	})
+}
+
+func (tc *EventsTest) BeforeAll(f *framework.F) {
+	e2eutil.WaitForLeader(f.T(), tc.Nomad())
+}
+
+func (tc *EventsTest) AfterEach(f *framework.F) {
+	nomadClient := tc.Nomad()
+	j := nomadClient.Jobs()
+
+	for _, id := range tc.jobIDs {
+		j.Deregister(id, true, nil)
+	}
```

Comment:
We should GC when we're done with each test (see an example in  [`vaultsecrets.go#L69-L70`](https://github.com/hashicorp/nomad/blob/master/e2e/vaultsecrets/vaultsecrets.go#L69-L70))